great_expectations/experimental/metric_repository/cloud_data_store.py

```diff
@@ -1,23 +1,80 @@
-from typing import TypeVar, Union
+from __future__ import annotations
 
-from typing_extensions import TypeAlias
+from typing import TYPE_CHECKING, Any, Dict, TypeVar, Union
+
+import pydantic
+from pydantic import BaseModel
 
 from great_expectations.compatibility.typing_extensions import override
+from great_expectations.core.http import create_session
 from great_expectations.experimental.metric_repository.data_store import DataStore
 from great_expectations.experimental.metric_repository.metrics import MetricRun
 
-StorableTypes: TypeAlias = Union[MetricRun,]  # TODO: are there better approaches?
+if TYPE_CHECKING:
+    from typing_extensions import TypeAlias
+
+    from great_expectations.data_context import CloudDataContext
+
+StorableTypes: TypeAlias = Union[MetricRun,]
 
 T = TypeVar("T", bound=StorableTypes)
 
 
+class PayloadData(BaseModel):
+    type: str
+    attributes: Dict[str, Any]
+
+    class Config:
+        extra = pydantic.Extra.forbid
+
+
+class Payload(BaseModel):
+    data: PayloadData
+
+    class Config:
+        extra = pydantic.Extra.forbid
+
+
 class CloudDataStore(DataStore[StorableTypes]):
+    """DataStore implementation for GX Cloud.
+
+    Uses JSON:API https://jsonapi.org/
+    """
+
+    @override
+    def __init__(self, context: CloudDataContext):
+        super().__init__(context=context)
+        self._session = create_session(
+            access_token=context.ge_cloud_config.access_token
+        )
+
+    def _map_to_url(self, value: StorableTypes) -> str:
+        if isinstance(value, MetricRun):
+            return "/metric-runs"
+
+    def _map_to_resource_type(self, value: StorableTypes) -> str:
+        if isinstance(value, MetricRun):
+            return "metric-run"
+
+    def _build_payload(self, value: StorableTypes) -> dict:
+        payload = Payload(
+            data=PayloadData(
+                type=self._map_to_resource_type(value),
+                attributes=value.dict(
+                    exclude={"metrics": {"__all__": {"__orig_class__"}}}
```

Comment:
`__orig_class__` is excluded because we don't use it and it makes checking for equality difficult.

Reply:
makes sense.